dbt_semantic_interfaces/objects/where_filter/filter_renderer.py

```diff
@@ -42,161 +47,86 @@ class FilterCallParameterSets:
-                dimension_call_parameter_sets.append(dimension_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-            def render_time_dimension_call(  # noqa: D
-                self, time_dimension_call_parameter_set: TimeDimensionCallParameterSet
-            ) -> str:
-                time_dimension_call_parameter_sets.append(time_dimension_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-            def render_entity_call(self, entity_call_parameter_set: EntityCallParameterSet) -> str:  # noqa: D
-                entity_call_parameter_sets.append(entity_call_parameter_set)
-                return FilterRenderer._DUMMY_PLACEHOLDER
-
-        FilterRenderer.render(
-            templated_filter_sql=templated_filter,
-            call_renderer=_DummyCallRenderer(),
-        )
-
-        return FilterCallParameterSets(
-            dimension_call_parameter_sets=tuple(dimension_call_parameter_sets),
-            time_dimension_call_parameter_sets=tuple(time_dimension_call_parameter_sets),
-            entity_call_parameter_sets=tuple(entity_call_parameter_sets),
-        )
-
-    @staticmethod
-    def render(
-        templated_filter_sql: str,
-        call_renderer: FilterFunctionCallRenderer,
-    ) -> str:
-        """Render the templated SQL according to the supplied functions. See class docstring for an example.
-
-        Args:
-            templated_filter_sql: the templated SQL to render
-            call_renderer: Specifies how to render calls like dimension(...) in the template.
-
-        Returns:
-            The templated SQL rendered according to the supplied functions.
-        """
-
-        def _dimension_helper(dimension_name: str, entity_path: Sequence[str] = ()) -> str:
+        def _dimension_call(dimension_name: str, entity_path: Sequence[str] = ()) -> str:
             """Gets called by Jinja when rendering {{ dimension(...) }}."""
-            return call_renderer.render_dimension_call(
+            dimension_call_parameter_sets.append(
                 DimensionCallParameterSet(
                     dimension_reference=DimensionReference(element_name=dimension_name),
                     entity_path=tuple(EntityReference(element_name=arg) for arg in entity_path),
                 )
             )
+            return ParseToCallParameterSets._DUMMY_PLACEHOLDER
 
-        def _time_dimension_helper(
+        def _time_dimension_call(
             time_dimension_name: str, time_granularity_name: str, entity_path: Sequence[str] = ()
         ) -> str:
             """Gets called by Jinja when rendering {{ time_dimension(...) }}."""
-            return call_renderer.render_time_dimension_call(
+            time_dimension_call_parameter_sets.append(
                 TimeDimensionCallParameterSet(
                     time_dimension_reference=DimensionReference(element_name=time_dimension_name),
```

Comment:
It makes sense that we use `DimensionReference` instead of `TimeDimensionReference` here as we've typed the `TimeDimensionCallParameterSet` to have a `DimensionReference`. However, why does `TimeDimensionCallParamterSet` have a `DimensionReference` instead of a `TimeDimensionReference`?

Reply:
Oversight - I'll update to `TimeDimensionReference`.